common/scala/build.gradle

```diff
@@ -47,6 +47,10 @@ dependencies {
     compile 'io.kamon:kamon-statsd_2.11:0.6.7'
     //for mesos
     compile 'com.adobe.api.platform.runtime:mesos-actor:0.0.7'
+
+    compile 'io.reactivex:rxscala_2.11:0.26.5'
+    compile 'io.reactivex:rxjava-reactive-streams:1.2.1'
+    compile 'com.microsoft.azure:azure-cosmosdb:1.0.0'
```

Comment:
Below is the dependency tree being pulled in (see [here][1] for full tree)


```
+--- io.reactivex:rxscala_2.11:0.26.5
     |    +--- org.scala-lang:scala-library:2.11.8 -> 2.11.12
     |    \--- io.reactivex:rxjava:1.2.4 -> 1.3.3
     +--- io.reactivex:rxjava-reactive-streams:1.2.1
     |    +--- io.reactivex:rxjava:1.2.2 -> 1.3.3
     |    \--- org.reactivestreams:reactive-streams:1.0.0 -> 1.0.2
     \--- com.microsoft.azure:azure-cosmosdb:1.0.0
          +--- com.fasterxml.jackson.core:jackson-databind:2.9.4 (*)
          +--- com.fasterxml.uuid:java-uuid-generator:3.1.4
          +--- org.json:json:20140107
          +--- commons-io:commons-io:2.5 -> 2.6
          +--- com.github.davidmoten:rxjava-extras:0.8.0.11
          |    \--- io.reactivex:rxjava:1.3.3
          +--- io.reactivex:rxjava:1.3.3
          +--- io.reactivex:rxjava-string:1.1.1
          |    \--- io.reactivex:rxjava:1.2.3 -> 1.3.3
          +--- io.reactivex:rxnetty:0.4.20
          |    +--- io.reactivex:rxjava:1.2.2 -> 1.3.3
          |    +--- io.netty:netty-codec-http:4.1.5.Final -> 4.1.20.Final
          |    |    \--- io.netty:netty-codec:4.1.20.Final
          |    |         \--- io.netty:netty-transport:4.1.20.Final
          |    |              +--- io.netty:netty-buffer:4.1.20.Final
          |    |              |    \--- io.netty:netty-common:4.1.20.Final
          |    |              \--- io.netty:netty-resolver:4.1.20.Final
          |    |                   \--- io.netty:netty-common:4.1.20.Final
          |    +--- io.netty:netty-handler:4.1.5.Final -> 4.1.20.Final
          |    |    +--- io.netty:netty-buffer:4.1.20.Final (*)
          |    |    +--- io.netty:netty-transport:4.1.20.Final (*)
          |    |    \--- io.netty:netty-codec:4.1.20.Final (*)
          |    \--- org.slf4j:slf4j-api:1.7.6 -> 1.7.25
          +--- io.netty:netty-codec-http:4.1.20.Final (*)
          +--- io.netty:netty-handler:4.1.20.Final (*)
          +--- io.netty:netty-transport:4.1.20.Final (*)
          +--- org.slf4j:slf4j-api:1.7.6 -> 1.7.25
          \--- org.apache.commons:commons-lang3:3.5
```  

Further did a diff on jars being packaged in controller.tar

```diff
30a31
> azure-cosmosdb-1.0.0.jar
34a36
> commons-lang3-3.5.jar
41,43c43,45
< jackson-annotations-2.7.0.jar
< jackson-core-2.7.7.jar
< jackson-databind-2.7.7.jar
---
> jackson-annotations-2.9.0.jar
> jackson-core-2.9.4.jar
> jackson-databind-2.9.4.jar
46c48
< java-uuid-generator-3.1.3.jar
---
> java-uuid-generator-3.1.4.jar
52a55
> json-20140107.jar
69a73,79
> netty-buffer-4.1.20.Final.jar
> netty-codec-4.1.20.Final.jar
> netty-codec-http-4.1.20.Final.jar
> netty-common-4.1.20.Final.jar
> netty-handler-4.1.20.Final.jar
> netty-resolver-4.1.20.Final.jar
> netty-transport-4.1.20.Final.jar
78a89,94
> rxjava-1.3.3.jar
> rxjava-extras-0.8.0.11.jar
> rxjava-reactive-streams-1.2.1.jar
> rxjava-string-1.1.1.jar
> rxnetty-0.4.20.jar
> rxscala_2.11-0.26.5.jar

```      

So following existing dependency versions are getting updated

1. jackson - 2.7 -> 2.9 - (minor). So far jackson was being used by kubernetes-client
2. java-uuid-generator - 3.1.3 -> 3.1.4 (micro) 

Following new set of library being included

1. netty - 4.1.20 - 7 jars
2. rxjava  - 6
3. commons-lang3
4. json - This would get replaced with jackson post Azure/azure-cosmosdb-java#29

In total post this change 

1. Number of jars in controller increase from 92 -> 108
2. Size increases from 89M -> 94M     

[1]: https://gist.github.com/chetanmeh/ab354bb8c320020b8c3d4233bf8f3f62

Reply:
Only concern is the jump in netty version - when I build controller.tar from your branch I get both netty versions:
```tnorris@tnorris-macOS ~/git/forks/adobe-apiplatform/incubator-openwhisk-mac -  (chetanmeh-artifact-store-cosmos) $ tar -tvf core/controller/build/distributions/controller.tar  | grep netty
-rw-r--r--  0 0      0      413674 May 18 13:27 controller/lib/rxnetty-0.4.20.jar
-rw-r--r--  0 0      0      562559 May 18 13:27 controller/lib/netty-codec-http-4.1.22.Final.jar
-rw-r--r--  0 0      0      374389 May 18 13:27 controller/lib/netty-handler-4.1.22.Final.jar
-rw-r--r--  0 0      0      316316 May 18 13:27 controller/lib/netty-codec-4.1.22.Final.jar
-rw-r--r--  0 0      0      456605 May 18 13:27 controller/lib/netty-transport-4.1.22.Final.jar
-rw-r--r--  0 0      0      270720 May 18 13:27 controller/lib/netty-buffer-4.1.22.Final.jar
-rw-r--r--  0 0      0       32276 May 18 13:27 controller/lib/netty-resolver-4.1.22.Final.jar
-rw-r--r--  0 0      0      575748 May 18 13:27 controller/lib/netty-common-4.1.22.Final.jar
-rw-r--r--  0 0      0     1292696 Sep 13  2017 controller/lib/netty-3.10.6.Final.jar```